dotnet-desktop-guide/net/winforms/printing/how-to-print-in-windows-forms-using-print-preview.md

```diff
@@ -0,0 +1,64 @@
+ms.date: "05/02/2022"
+dev_langs: 
+  - "csharp"
+  - "vb"
+helpviewer_keywords: 
+  - "printing [Windows Forms], using print preview"
+  - "printing [Windows Forms], with print preview"
+  - "print preview"
+ms.custom: devdivchpfy22
+---
+
+# How to print in Windows forms using print preview
+
+It's common in Windows Forms programming to offer print preview in addition to printing services. An easy way to add print preview services to your application is to use a <xref:System.Windows.Forms.PrintPreviewDialog> control in combination with the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event-handling logic for printing a file.
+
+## To preview a text document with a PrintPreviewDialog control
+
+01. Add a <xref:System.Windows.Forms.PrintPreviewDialog> and <xref:System.Drawing.Printing.PrintDocument>.
+
+01. Add two strings to your form.
+
+     :::code language="csharp" source="snippets/how-to-print-in-windows-forms-using-print-preview/csharp/Form1.cs" id="string_declaration":::
+
+     :::code language="vb" source="snippets/how-to-print-in-windows-forms-using-print-preview/vb/Form1.vb" id="string_declaration":::
+
+01. Set the <xref:System.Drawing.Printing.PrintDocument.DocumentName%2A> property to the document you wish to print, and open and read the document's contents to the string you added previously.
+
+     :::code language="csharp" source="snippets/how-to-print-in-windows-forms-using-print-preview/csharp/Form1.cs" id="open_and_read_document_contents_the_string":::
+
+     :::code language="vb" source="snippets/how-to-print-in-windows-forms-using-print-preview/vb/Form1.vb" id="open_and_read_document_contents_the_string":::
+
+01. As you would for printing the document, in the <xref:System.Drawing.Printing.PrintDocument.PrintPage> event handler, use the <xref:System.Drawing.Printing.PrintPageEventArgs.Graphics%2A> property of the <xref:System.Drawing.Printing.PrintPageEventArgs> class and the file contents to calculate lines per page and render the document's contents. After each page is drawn, check to see if it's the last page, and set the <xref:System.Drawing.Printing.PrintPageEventArgs.HasMorePages%2A> property of the `PrintPageEventArgs` accordingly. The `PrintPage` event is raised until `HasMorePages` is `false`. When the document has finished rendering, reset the string to be rendered. Also, ensure that the `PrintPage` event is associated with its event-handling method.
+
+    > [!NOTE]
+    > If you have implemented printing in your application, you may have already completed step 2 and 3.
+
+     In the following code example, the event handler is used to print the "testPage.txt" file in the same font used on the form.
+
+     :::code language="csharp" source="snippets/how-to-print-in-windows-forms-using-print-preview/csharp/Form1.cs" id="print_file_using_event_handler":::
+
+     :::code language="vb" source="snippets/how-to-print-in-windows-forms-using-print-preview/vb/Form1.vb" id="print_file_using_event_handler":::
```

Comment:
Indentation is wrong here, they seem to be indented 5 spaces when they should be 4. I noticed a lot of the other indentation is suffering from the same issue. Can you please double check and fix them throughout the document? Thanks!

```suggestion
    > [!NOTE]
    > If you have implemented printing in your application, you may have already completed step 2 and 3.

    In the following code example, the event handler is used to print the "testPage.txt" file in the same font used on the form.

    :::code language="csharp" source="snippets/how-to-print-in-windows-forms-using-print-preview/csharp/Form1.cs" id="print_file_using_event_handler":::

    :::code language="vb" source="snippets/how-to-print-in-windows-forms-using-print-preview/vb/Form1.vb" id="print_file_using_event_handler":::
```